packages/loader/container-loader/src/retriableDocumentStorageService.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { CreateContainerError } from "@fluidframework/container-utils";
+import { IDocumentStorageService } from "@fluidframework/driver-definitions";
+import { canRetryOnError, DocumentStorageServiceProxy } from "@fluidframework/driver-utils";
+import { ISnapshotTree, IVersion } from "@fluidframework/protocol-definitions";
+import { Container } from "./container";
+import { emitThrottlingWarning, getRetryDelayFromError } from "./deltaManager";
+
+export class RetriableDocumentStorageService extends DocumentStorageServiceProxy {
+    private shouldRetry = true;
+    constructor(
+        internalStorageService: IDocumentStorageService,
+        private readonly container: Container,
+    ) {
+        super(internalStorageService);
+    }
+
+    public stopRetry() {
+        this.shouldRetry = false;
+    }
+
+    public async getSnapshotTree(version?: IVersion): Promise<ISnapshotTree | null> {
+        return this.readWithRetry(async () => this.internalStorageService.getSnapshotTree(version));
+    }
+
+    public async read(blobId: string): Promise<string> {
```

Comment:
you are missing newly added API readString().
How do we ensure we are not missing any future APIs that get added? I.e. should we maybe not have base class that implements all these APIs? Not sure if it outweighs the benefits of simplicity

Reply:
Added the api. I think we should go for base class for simplicity as then we don't need to replicate all apis. Such api(readString) are not added frequently so those I think we should go for simplicity here.